enhancements/network/sdn-live-migration.md

```diff
@@ -0,0 +1,442 @@
+cluster MTU by 50 bytes following this instruction:
+[changing-cluster-network-mtu](https://docs.openshift.com/container-platform/4.13/networking/changing-cluster-network-mtu.html).
+
+### The Traffic Path
+
+#### Packets going from OpenShift SDN to OVN Kubernetes
+
+On the SDN side, it doesn't need to know if the peer node is a SDN node or an
+OVN node. We reuse the existing VXLAN tunnel rules on the SDN side.
+- Egress NetworkPolicy rules and service proxy happen as normal.
+- When the packet reaches table 90, it will hit a "send via vxlan" rule that was
+  generated based on a HostSubnet object.
+
+On the OVN side:
+- OVN accepts the packet via the VXLAN tunnel, ignores the VNID set by SDN, and
+  then just routes it normally.
+- Ingress NetworkPolicy processing will happen when the packet reaches the
+  destination pod's switch port, just like normal.
+  - Our NetworkPolicy rules are all based on IP addresses, not "logical input
+    port", etc., so it doesn't matter that the packets came from outside OVN and
+    have no useful OVN metadata.
+
+#### Packets going from OVN Kubernetes to OpenShift SDN
+
+On the OVN side:
+- The packet just follows the same path as the hybrid overlay.
+
+On the SDN side:
+- We have to change ingress NetworkPolicy processing to be based entirely on pod
+  IPs rather than using namespace VNIDs since packets from OVN nodes won't have
+  the VNID set. There is already code to generate the rules that way, though,
+  because egress NetworkPolicy already works that way.
+
+### Workflow Description
+
+1. The admin kicks off the migration process by updating the custom resource
+   `network.config`:
+   
+   ```bash
+   $ oc patch Network.config.openshift.io cluster --type='merge' --patch '{"metadata":{"annotations":{"network.openshift.io/live-migration":""}},"spec":{"networkType":"OVNKubernetes"}}'
+   ```
+   
+   CNO will set the `LiveMigrationProgressing` condition to TRUE in the `status` of
+   the network.config CR. If there are any unsupported features (refer to the
+   [Limitations](#limitations) section) enabled in the cluster. CNO will
+   set an the `LiveMigrationProgressing` condition to `False` with warning message
+   and won't start the live migration. User shall disable these unsupported
+   feature before starting the live migration.
+
+2. CNO will check the current hardware MTU and the cluster MTU to see if there
+   is enough free space for OVN-K overlay overhead. If not, sets the
+   `.status.migration.mtu` of network.config CR with MTU of OVN-k, e.g.:
+   - `.status.migration.mtu.network.from` 1450 `to` 1400
+   - `.status.migration.mtu.machine.from` 1500 `to` 1500
+   
+   This will trigger MCO to apply a new machine config to each MCP and reboots
+   all the nodes.
+
+3. CNO watches the MCP, wait until the MCs with routable MTU is applied to all
+   the nodes. CNO then sets the `LiveMigrationRoutableMTUApplied` condition to
```

Comment:
In this design, we try to make the `network.config` API be declarative. The `network.operator` API still works in the old way. So it would need extra effort to make the MTU migration be declarative and automated.

Reply:
I was not suggesting to cover a declarative MTU migration in this spec, but rather think ahead of time whether some statuses should be shared in case that ever happens in the future.

In any case though, `RoutableMTU` looks as an implantation detail as well. Could it just read `LiveMigrationMTUApplied` or `LiveMigrationMTUReady` or something like that?